src/ripple/shamap/SHAMap.h

```diff
@@ -182,10 +182,9 @@ class SHAMap
     
     bool getRootNode (Serializer & s, SHANodeFormat format) const;
     std::vector<uint256> getNeededHashes (int max, SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (uint256 const& hash, Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
+    SHAMapAddNode addRootNode (boost::optional<uint256> const& hash,
```

Comment:
`optional` seems the simplest way to be able to indicate that the caller doesn't have a `hash`. If there's a better way, I'm open to it.